src/bridge/components/StatelessTransaction.js

```diff
@@ -85,7 +85,7 @@ class StatelessTransaction extends React.Component {
 
   submit(){
     const { props, state } = this
-    sendSignedTransaction(props.web3.value, state.stx)
+    sendSignedTransaction(props.web3.value, state.stx, props.setTxnConfirmations)
```

Comment:
I just noticed that `web3` isn't being unwrapped here.  I.e., `web3` could be `Nothing`, such that `web3.value` would be `undefined`.  It should be unwrapped in the standard fashion; something like:

```
const web3 = props.web3.matchWith({
  Just: w3 => w3.value,
  Nothing: _ => { throw BRIDGE_ERROR.MISSING_WEB3 }
})

sendSignedTransaction(web3, ...)
```

That folktale lets one just wing it on the `value` property is pretty lousy, but alas, such is the nature of JS.

(This isn't actually related to your change @pkova -- I just happened to spot it while scrolling through the diff.)

Reply:
Is it not ok to use `.value` directly if pre-condition logic (ie whatever you need to do to make `submit()` accessible) already guarantees/checks for `Just`? (Of course, figuring that out as the reader isn't always trivial, but "trust the author"? (^: )

Reply:
The reason you want to be pedantic and do it literally everywhere is that it's easy to change the code (perhaps in a refactoring), remove the precondition, but forget to change the yolo `.value` attempt.  It's a recipe for introducing annoying bugs eventually.

Better would probably be to do the unwrapping once, in one place, consistently.  This is what I was thinking when I mentioned unwrapping all the necessary props in the constructor, or at least all in one place, and then to use the unwrapped values elsewhere in the component.  Maybe something like:

```
class Foo extends React.Component {
  constructor(props) {
    super(props)

    this.web3 = props.web3.matchWith({
      Nothing: _ => throw { BRIDGE_ERROR.MISSING_WEB3 },
      Just: w3 => w3.value
    })
  }

  render() {
    const web3 = this.web3

    return(
      <div>{ web3 }</div>
    )
  }
}
```

Reply:
Definitely in favor of unwrapping once in the constructor, yes. I'll put that on my informal refactoring list.

Reply:
I don't like Maybe. Using it has the upside of 'forcing you to check' for missing values, but the tradeoff is that changes that should be non-breaking become breaking changes. If we have a function 

`foo :: x -> Maybe y` 

and we manage to improve it so that we can always return y, eg. 

`foo :: x -> y` 

existing callers of our function break. Nullable types a la Kotlin or Typescript seem better. [Rich explains this far better than I ever could.](https://github.com/matthiasn/talk-transcripts/blob/master/Hickey_Rich/MaybeNot.md)

Don't think there's a solution for this in js-land, though!

Reply:
I'm a Haskeller, so Rich's talk triggers the @#&! out of me.  But what I can certainly agree on is that, yes, there's no satisfactory solution for this in JS-land. ;-)